packages/synapse-react-client/src/components/EntityViewScopeEditor/EntityViewScopeEditorModal.test.tsx

```diff
@@ -0,0 +1,235 @@
+import mockTableEntityData, {
+  mockTableEntity,
+} from '../../mocks/entity/mockTableEntity'
+import {
+  ENTITY_VIEW_TYPE_MASK_FILE,
+  ENTITY_VIEW_TYPE_MASK_FOLDER,
+  ENTITY_VIEW_TYPE_MASK_PROJECT,
+  ENTITY_VIEW_CONCRETE_TYPE_VALUE,
+  Entity,
+} from '@sage-bionetworks/synapse-types'
+import { MOCK_ACCESS_TOKEN } from '../../mocks/MockSynapseContext'
+import { SynapseClientError } from '../../utils'
+import { mockProjectViewEntity } from '../../mocks/entity/mockProjectView'
+
+const mockTableEntityInstance = mockTableEntityData.entity
+
+jest.mock('../EntityViewScopeEditor/EntityViewScopeEditor', () => ({
+  __esModule: true,
+  default: jest.fn(() => <div data-testid={'EntityViewScopeEditorrMocked'} />),
+}))
+const mockEntityViewScopeEditor = jest.mocked(EntityViewScopeEditor)
+
+function renderComponent(props: EntityViewScopeEditorModalProps) {
+  return render(<EntityViewScopeEditorModal {...props} />, {
+    wrapper: createWrapper(),
+  })
+}
+
+const mockUpdateEntity = jest.spyOn(SynapseClient, 'updateEntity')
+
+describe('EntityViewScopeEditorModal tests', () => {
+  const mockOnCancel = jest.fn()
+  const mockOnUpdate = jest.fn()
+
+  beforeEach(() => {
+    jest.spyOn(SynapseClient, 'getEntity').mockResolvedValue({
+      ...mockTableEntity,
+      scopeIds: ['syn123'],
+      viewTypeMask: ENTITY_VIEW_TYPE_MASK_FILE | ENTITY_VIEW_TYPE_MASK_FOLDER,
+      concreteType: ENTITY_VIEW_CONCRETE_TYPE_VALUE,
+    } as Entity)
+  })
+
+  it('displays the correct scope editor and mask editor after initial render', async () => {
+    jest.clearAllMocks()
+    renderComponent({
+      entityId: mockTableEntity.id,
+      open: true,
+      onCancel: mockOnCancel,
+      onUpdate: mockOnUpdate,
+    })
+
+    const dialog = screen.getByRole('dialog')
+    expect(
+      within(dialog).getByRole('heading', { name: 'Edit Scope' }),
+    ).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Cancel' })).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Save' })).toBeVisible()
+
+    const filesButton = within(dialog).getByRole('checkbox', { name: 'Files' })
+    const foldersButton = within(dialog).getByRole('checkbox', {
+      name: 'Folders',
+    })
+    const tablesButton = within(dialog).getByRole('checkbox', {
+      name: 'Tables',
+    })
+    const datasetsButton = within(dialog).getByRole('checkbox', {
+      name: 'Datasets',
+    })
+
+    const checkboxes = screen.getAllByRole('checkbox')
+    expect(checkboxes).toHaveLength(4)
+
+    expect(filesButton).toBeVisible()
+    expect(foldersButton).toBeVisible()
+    expect(tablesButton).toBeVisible()
+    expect(datasetsButton).toBeVisible()
+
+    await waitFor(() => {
+      expect(filesButton).toBeChecked()
+      expect(foldersButton).toBeChecked()
+      expect(tablesButton).not.toBeChecked()
+      expect(datasetsButton).not.toBeChecked()
+    })
+  })
+
+  it('display error for invalid entity', async () => {
```

Comment:
Rename test for clarity; there's no validity issue with the entity here
```suggestion
  it('displays error response returned by update call', async () => {
```